api/mfx_dispatch/linux/mfxloader.cpp

```diff
@@ -177,23 +180,35 @@ mfxStatus LoaderCtx::Init(mfxInitParam& par)
     return MFX_ERR_UNSUPPORTED;
   }
 
+  eMFXHWType platform = MFX_HW_UNKNOWN;
+  auto devices = get_devices();
+  if (devices.size()) {
+    platform = devices[0].platform;
```

Comment:
You sorted deive list per platform, so MFX_HW_UNKNOWN will be the [0] one. And you will load VPL while we need to load MSDK runtime. Please, fix.

Reply:
Fixed. Now devices[devices.size() - 1] is used to automatically select legacy path if MSDK and VPL platform are detected at the same time